glusterd2/commands/peers/editpeer.go

```diff
@@ -104,10 +104,20 @@ func txnPeerEdit(c transaction.TxnCtx) error {
 		return err
 	}
 
+	reqMetadataSize := req.MetadataSize()
+	if reqMetadataSize > maxMetadataSizeLimit {
+		return gderrors.ErrMetadataSizeOutOfBounds
+	}
+
 	for k, v := range req.Metadata {
 		peerInfo.Metadata[k] = v
 	}
 
+	metadataSize := peerInfo.MetadataSize()
```

Comment:
Use the function call directly in if condition, there's no need for extra variable.